packages/flutter_tools/lib/src/base/args.dart

```diff
@@ -0,0 +1,35 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:args/args.dart';
+
+/// A typed wrapper for the flutter command [ArgResults].
+class TypedArgResults {
+  TypedArgResults(this._argResults);
+
+  final ArgResults _argResults;
+
+  /// Returns the value of the command line flag `name`.
+  bool readFlag(String name) => _argResults[name] as bool;
```

Comment:
What do you think about:

```dart
Map<String, bool> _flags;
Map<String, bool> get flags {
  if (_flags == null) {
    _flags = Map.fromEntries(_argResults.entries.where((e) => e.value is bool);
  }
  return _flags;
}
```

Reply:
My hope is that something like readFlag or flag is added by the args package (https://github.com/dart-lang/args/issues/95) and then we could drop the wrapper entirely.